projects/canopy/src/lib/skeleton/skeleton.directive.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Directive,
+  ElementRef,
+  Renderer2,
+  Input,
+  HostBinding,
+  OnDestroy,
+} from '@angular/core';
+
+@Directive({
+  selector: '[lgSkeleton]',
+})
+export class LgSkeletonDirective implements OnDestroy {
+  private changes: MutationObserver;
+
+  @Input() lgSkeletonWidth: string = null;
+  @Input() lgSkeletonHeight: string = null;
+  @Input() lgSkeletonRightAlign = false;
+
+  @HostBinding('class.lg-skeleton') skeletonClass = true;
+  @HostBinding('class.lg-skeleton--right')
+  get alignClass() {
+    return this.lgSkeletonRightAlign;
+  }
+
+  @HostBinding('style.width')
+  get backgroundWidth(): string {
+    return this.lgSkeletonWidth ? `${this.lgSkeletonWidth}em` : '100%';
+  }
+
+  @HostBinding('style.height')
+  get backgroundHeight(): string {
+    return this.lgSkeletonHeight ? `${this.lgSkeletonHeight}em` : 'auto';
+  }
+
+  constructor(private renderer: Renderer2, private hostElement: ElementRef) {
+    this.changes = new MutationObserver(this.observeChanges);
+
+    this.changes.observe(this.hostElement.nativeElement, {
+      subtree: true,
+      characterData: true,
+    });
+  }
+
+  observeChanges = (mutations: Array<MutationRecord>) => {
+    mutations.forEach((mutation: MutationRecord) => {
```

Comment:
I wondered if we could set a boolean property for `hasContent` or similar. Then the behaviour for removing the classes could be done in the host binding as well. Could be a bit less code?
```ts
mutations.forEach((mutation: MutationRecord) => {
  this.hasContent = mutation.target.textContent.trim() !== ''
}); 
```


```ts
get alignClass() {
  return this.lgSkeletonRightAlign && !this.hasContent;
}
```

Reply:
Yeah, I like that. Will make that change

Reply:
think this one might need resolving still, if you think it's a go-er?